### PR TITLE
tests(core): add test cases for simdjson yielding

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Add more test cases for #13421.

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

File: spec/01-unit/31-simdjson/02-yield_spec.lua [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    local parser = simdjson.new(true)
+    assert(parser)
+
+    local str = parser:encode(test_obj)
+
+    parser:destroy()
+
+    assert(str)
+    assert(type(str) == "string")
```
Comment (outsinre) [inline spec/01-unit/31-simdjson/02-yield_spec.lua:47]: How about adding something like this?

```lua
assert(string.find(str, [=["str":"aaa"]=]))
```

File: spec/01-unit/31-simdjson/02-yield_spec.lua [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    local parser = simdjson.new(true)
+    assert(parser)
+
+    local obj = parser:decode(test_str)
+
+    parser:destroy()
+
+    assert(obj)
+    assert(type(obj) == "table")
```
Comment (outsinre) [inline spec/01-unit/31-simdjson/02-yield_spec.lua:78]: How about adding an assert?

```lua
assert.equal(obj[1000], 1000)
```

**Review by outsinre — COMMENTED** [review]
LGTM

**Review by outsinre — APPROVED** [review]

File: spec/01-unit/31-simdjson/02-yield_spec.lua [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    assert.equal(obj[1], 1)
+    assert.equal(obj[1000], 1000)
+
+    assert.spy(spy_ngx_sleep).was_called(1)       -- yield once
```
Comment (StarlightIbuki) [inline spec/01-unit/31-simdjson/02-yield_spec.lua:86]: Do we need to specify how many times it yields?

File: spec/01-unit/31-simdjson/02-yield_spec.lua [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  after_each(function()
+    ngx.sleep = orig_ngx_sleep  -- luacheck: ignore
+    package.loaded["resty.simdjson"] = nil
+    package.loaded["resty.simdjson.decoder"] = nil
+    package.loaded["resty.simdjson.encoder"] = nil
+  end)
+
+
+  it("enabled when encoding", function()
```
Comment (StarlightIbuki) [inline spec/01-unit/31-simdjson/02-yield_spec.lua:37]: We can use a loop to iterate over 4 combinations

File: spec/01-unit/31-simdjson/02-yield_spec.lua (hunk 3 above)
Comment (chronolaw) [reply spec/01-unit/31-simdjson/02-yield_spec.lua:86]: In this case it only yield once.

File: spec/01-unit/31-simdjson/02-yield_spec.lua (hunk 4 above)
Comment (chronolaw) [reply spec/01-unit/31-simdjson/02-yield_spec.lua]: good suggestion, done.

**Review by StarlightIbuki — APPROVED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**